src/ILCompiler.Compiler/src/Compiler/CompilationModuleGroup.cs

```diff
@@ -27,8 +27,36 @@ protected CompilationModuleGroup(CompilerTypeSystemContext typeSystemContext, IC
             _rootProvider = rootProvider;
         }
 
-        public abstract bool IsTypeInCompilationGroup(TypeDesc type);
-        public abstract bool IsMethodInCompilationGroup(MethodDesc method);
+        /// <summary>
+        /// If true, "type" is in the set of input assemblies being compiled
+        /// </summary>
+        public abstract bool ContainsType(TypeDesc type);
+        /// <summary>
+        /// If true, "method" is in the set of input assemblies being compiled
+        /// </summary>
+        public abstract bool ContainsMethod(MethodDesc method);
+        /// <summary>
+        /// If true, it's possible for "type" to be generated in multiple modules independently and should be shared
+        /// </summary>
+        public abstract bool ShouldShareAcrossModules(TypeDesc type);
+        /// <summary>
+        /// If true, it's possible for "method" to be generated in multiple modules independently and should be shared
+        /// </summary>
+        public abstract bool ShouldShareAcrossModules(MethodDesc method);
+        /// <summary>
+        /// If true, all code is compiled into a single module
+        /// </summary>
+        public abstract bool IsSingleFileCompilation();
+        /// <summary>
+        /// Roots a type; all its methods will be compiled and an EEType will be produced
+        /// </summary>
+        public abstract void AddCompilationRoot(TypeDesc type);
```

Comment:
You're right - thanks for keeping me honest :)  I believe it's not needed anymore (I'll have a better idea of that when I write some more complicated test cases). EETypeVTableNode should now ensure that any virtual overrides for a shared type will be added to the graph.  For non-virtuals, since each method gets a COMDAT section, we don't even need to compile all methods of a shared type. Each module will contribute the methods it uses.